Mathlib/Probability/Kernel/Composition/IntegralCompProd.lean

```diff
@@ -261,4 +278,185 @@ theorem setIntegral_compProd_univ_left (f : β × γ → E) {t : Set γ} (ht : M
+    exact κ.measurable_coe ((s n).measurableSet_fiber _)
+  · rw [tendsto_pi_nhds]; intro x
+    by_cases hfx : Integrable f (κ x)
+    · simp only [mem_setOf_eq, hfx, indicator_of_mem, f']
+      apply tendsto_integral_approxOn_of_measurable_of_range_subset _ hfx
+      exact subset_rfl
+    · simp [f', hfx, integral_undef]
+
+theorem _root_.MeasureTheory.AEStronglyMeasurable.integral_kernel_comp [NormedSpace ℝ E]
+    ⦃f : γ → E⦄ (hf : AEStronglyMeasurable f ((η ∘ₖ κ) a)) :
+    AEStronglyMeasurable (fun x ↦ ∫ y, f y ∂η x) (κ a) :=
+  ⟨fun x ↦ ∫ y, hf.mk f y ∂η x, hf.stronglyMeasurable_mk.integral_kernel, by
+    filter_upwards [ae_ae_of_ae_comp hf.ae_eq_mk] with _ hx using integral_congr_ae hx⟩
+
+theorem _root_.MeasureTheory.AEStronglyMeasurable.comp {δ : Type*} [TopologicalSpace δ]
+    {f : γ → δ} (hf : AEStronglyMeasurable f ((η ∘ₖ κ) a)) :
+    ∀ᵐ x ∂κ a, AEStronglyMeasurable f (η x) := by
+  filter_upwards [ae_ae_of_ae_comp hf.ae_eq_mk] with x hx using
+    ⟨hf.mk f, hf.stronglyMeasurable_mk, hx⟩
+
+/-! ### Integrability with respect to composition -/
+
+theorem hasFiniteIntegral_comp_iff ⦃f : γ → E⦄ (hf : StronglyMeasurable f) :
+    HasFiniteIntegral f ((η ∘ₖ κ) a) ↔
+    (∀ᵐ x ∂κ a, HasFiniteIntegral f (η x)) ∧ HasFiniteIntegral (fun x ↦ ∫ y, ‖f y‖ ∂η x) (κ a) := by
+  simp_rw [hasFiniteIntegral_iff_enorm, lintegral_comp _ _ _ hf.enorm]
+  simp_rw [integral_eq_lintegral_of_nonneg_ae (ae_of_all _ fun y ↦ norm_nonneg _)
+      hf.norm.aestronglyMeasurable, enorm_eq_ofReal toReal_nonneg, ofReal_norm_eq_enorm]
+  have : ∀ {p q r : Prop} (_ : r → p), (r ↔ p ∧ q) ↔ p → (r ↔ q) := fun h ↦ by
+    rw [← and_congr_right_iff, and_iff_right_of_imp h]
+  rw [this]
+  · intro h
+    rw [lintegral_congr_ae]
+    filter_upwards [h] with x hx
+    rw [ofReal_toReal]
+    rwa [← lt_top_iff_ne_top]
+  · exact fun h ↦ ae_lt_top hf.enorm.lintegral_kernel h.ne
+
+theorem hasFiniteIntegral_comp_iff' ⦃f : γ → E⦄ (hf : AEStronglyMeasurable f ((η ∘ₖ κ) a)) :
+    HasFiniteIntegral f ((η ∘ₖ κ) a) ↔
+    (∀ᵐ x ∂κ a, HasFiniteIntegral f (η x)) ∧ HasFiniteIntegral (fun x ↦ ∫ y, ‖f y‖ ∂η x) (κ a) := by
+  rw [hasFiniteIntegral_congr hf.ae_eq_mk, hasFiniteIntegral_comp_iff hf.stronglyMeasurable_mk]
+  refine and_congr (eventually_congr ?_) (hasFiniteIntegral_congr ?_)
+  · filter_upwards [ae_ae_of_ae_comp hf.ae_eq_mk.symm] with _ hx using
+      hasFiniteIntegral_congr hx
+  · filter_upwards [ae_ae_of_ae_comp hf.ae_eq_mk.symm] with _ hx using
+      integral_congr_ae (EventuallyEq.fun_comp hx _)
+
+theorem integrable_comp_iff ⦃f : γ → E⦄ (hf : AEStronglyMeasurable f ((η ∘ₖ κ) a)) :
+    Integrable f ((η ∘ₖ κ) a) ↔
+    (∀ᵐ y ∂κ a, Integrable f (η y)) ∧ Integrable (fun y ↦ ∫ z, ‖f z‖ ∂η y) (κ a) := by
+  simp only [Integrable, hf, hasFiniteIntegral_comp_iff' hf, true_and, eventually_and, hf.comp,
+    hf.norm.integral_kernel_comp]
+
+theorem _root_.MeasureTheory.Integrable.comp ⦃f : γ → E⦄ (hf : Integrable f ((η ∘ₖ κ) a)) :
```

Comment:
`comp` seems a bit too generic for a name. Perhaphs this?
```suggestion
theorem _root_.MeasureTheory.Integrable.ae_of_comp ⦃f : γ → E⦄ (hf : Integrable f ((η ∘ₖ κ) a)) :
```
The name for the compProd version is `compProd_mk_left_ae`, which is not great either :)